nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -661,9 +670,10 @@ where
                         )
                         .await;
 
-                        Self::prune_old_forks(&mut cryptarchia, &mut leader, relays.storage_adapter()).await;
-                        // We update cryptarchia state only after attempting to prune blocks, to avoid saving blocks that have been already successfully pruned.
-                        self.service_resources_handle.state_updater.update(Some(Self::State::from_cryptarchia(&cryptarchia, &leader)));
+                        // This will modify `previously_pruned_blocks` to include blocks which are not tracked by Cryptarchia anymore but have not been deleted from the persistence layer.
+                        Self::prune_forks(&mut cryptarchia, &mut leader, relays.storage_adapter(), &mut previously_pruned_blocks).await;
+                        // We dump the undeleted blocks to storage so they are picked up by the service again if it suddenly stops.
+                        self.service_resources_handle.state_updater.update(Some(Self::State::from_cryptarchia_and_unpruned_blocks(&cryptarchia, &leader, previously_pruned_blocks.clone())));
```

Comment:
not sure I get this point

Reply:
We include the blocks that failed to be deleted in the service state, if any, to allow for recovery in case of sudden shutdown, what we usually do for services anyway. Was the comment not clear or what?